src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -276,10 +274,8 @@ public List<Course> findAllActiveWithExercisesAndLecturesAndExamsForUser(User us
             course.setExercises(allExercises.stream().filter(ex -> ex.getCourseViaExerciseGroupOrCourseMember().getId().equals(course.getId())).collect(Collectors.toSet()));
             course.setExercises(exerciseService.filterExercisesForCourse(course, user));
             exerciseService.loadExerciseDetailsIfNecessary(course, user);
+            course.setExams(allExams.stream().filter(ex -> ex.getCourse().getId().equals(course.getId())).collect(Collectors.toSet()));
```

Comment:
We only retrieved exams with matching courses before. I don't think we need to filter again, do we?

Reply:
We retreived the exams for all courses, so we need to find the exams for the course we are currently looping over

Reply:
Ah, the start of the peek stream was collapsed. Then it makes sense, as you do it for each course. My bad.